hal/src/boron/cellular_hal.cpp

```diff
@@ -136,6 +136,13 @@ int cellular_device_info(CellularDevice* info, void* reserved) {
     CHECK_TRUE(mgr, SYSTEM_ERROR_UNKNOWN);
     const auto client = mgr->ncpClient();
     CHECK_TRUE(client, SYSTEM_ERROR_UNKNOWN);
+
+    client->lock();
+    SCOPE_GUARD({
+        client->unlock();
+    });
```

Comment:
There's a lock guard for NCP client classes: https://github.com/particle-iot/device-os/blob/mesh-develop/hal/network/ncp/ncp_client.h#L110

Reply:
Completely forgot about it. Fixed.